### ISSUE TITLE
json-schema-ref-parser dependency breaks node engine < 17

### ISSUE THREAD

**Issue by ashwins01** [exegesis-js/exegesis#423]
`"json-schema-ref-parser": "^7.0.0"` was changed to `"@apidevtools/json-schema-ref-parser": "^9.0.3"` and a breaking change which was meant only for v10 of the `json-schema-ref-parser` package has snuck its way into v9 which is breaking `firebase-tools` which uses `exegesis`.
Is there any way the version of `json-schema-ref-parser` be pinned to `9.1.0` in exegesis so other packages using exegesis will not break for node engines running on versions <17 ?

Few links for reference,
Versions list of `json-schema-ref-parser`: https://www.npmjs.com/package/@apidevtools/json-schema-ref-parser?activeTab=versions
Issue raised in `json-schema-ref-parser`: https://github.com/APIDevTools/json-schema-ref-parser/issues/298#issuecomment-1399198083
PR which broke `json-schema-ref-parser`: https://github.com/APIDevTools/json-schema-ref-parser/pull/295#issuecomment-1399098077
Commit in which `"json-schema-ref-parser": "^7.0.0"` was changed to `"@apidevtools/json-schema-ref-parser": "^9.0.3"`: https://github.com/exegesis-js/exegesis/commit/17303c94c45d824a692b4c6a13157dcdc6869b44

**TLDR:**
Version `9.1.1` of `json-schema-ref-parser` breaks internet, please pin this dependency's version to `9.1.0`

@jwalton TIA

**Comment by coffeebe4code** [comment]
Step over Kim Kardashian, this PR came through to break the internet :)

I can confirm also seeing the same issues. It at least sparked internal debate about finally upgrading to the LTS version of node.

**Comment by jwalton** [comment]
I'd _rather_ see this fixed with a json-schema-ref-parser@9.1.2.  Pinning dependencies like that is always a pain, because it potentially prevents you from picking up a security fix later on, and if 9.1.2 does come out, I'll have to revert the change.

If you're on npm version 8, you can use [package overrides](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides) to fix this:

```json
  "overrides": {
    "@apidevtools/json-schema-ref-parser": "9.1.0"
  }
```

If you're using yarn you can fix this with almost the same syntax but with "resolutions" instead of "overrides" (Why did these two package managers pick different keys for this? -_-)

```json
  "resolutions": {
    "@apidevtools/json-schema-ref-parser": "9.1.0"
  }
```

If you're on an older npm, or on a dev team where some people insist on using yarn and some insist on using npm, I believe you can fix this with `npm install @apidevtools/json-schema-ref-parser@9.1.0`, which will pin this version in your package.json, and that should fix it for exegesis too.

If none of that works for you, and you really need a new exegesis release, then let me know and I'll make it happen.  :)  But please try those first.
